JumpscaleLib/sal_zos/storage/StoragePool.py

```
@@ -228,13 +195,14 @@ def get_devices_and_status(self):
                     status = 'degraded'
                     pool_status = 'degraded'
 
-            device_map.append({
+            device = {
                 'device': device,
                 'partUUID': info['partuuid'] or '' if info else '',
                 'status': status,
-            })
+            }
 
-        return device_map, pool_status
+            return device, pool_status
+        return {}, ''
```

Shouldnt we raise an error here?

raise DeviceNotFoundError()